### PR TITLE
Make apple frameworks optional

### PR DESCRIPTION
**Pull request by matthewbauer**
Previously, you had to have the apple sdk frameworks downloaded to
build on “Darwin”. There are certain cases where this is not desired,
so an Autoconf conditional is added to check for their availability.
When they are not available, proctitle & fsevents are unavailable.

These frameworks are proprietary- owned and developed by Apple Inc.
They have never been released publically so we should not make
everyone use it in a core library like libuv.

Also could not get hrtime test to work on my macOS - so had to disable it.

### CODE REVIEW

**Comment by saghul** [comment]
What kind of environment doesn't have those? PureDarwin?

**Comment by matthewbauer** [comment]
This is for Nixpkgs which kind of acts similarly to PureDarwin. We do have access to these SDKs but since they are unfree we would like to limit their usage as much as possible especially for nonessential features.

**Comment by santigimeno** [comment]
What error are you getting with the hrtime test?

File: src/unix/darwin-proctitle.c [hunk 1]
```diff
@@ -26,9 +26,7 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include <TargetConditionals.h>
-
-#if !TARGET_OS_IPHONE
+#if HAVE_APPLICATIONSERVICES_APPLICATIONSERVICES_H
```
Comment (saghul) [inline src/unix/darwin-proctitle.c]: this won't work on a GYP / CMake build I think.

**Comment by saghul** [comment]
I see. I'm kinda -1 here. Following that logic we'd also make IOCP optional on Windows, because it's also non-free.

**Comment by matthewbauer** [comment]
Ok - I hadn't realized there was another build system. That could make things trickier. It's definitely okay to not merge - this is just what we will be using & it's a fairly straightforward patch. Note that without something like this, the fsevents tests will still fail on iOS.

**Comment by matthewbauer** [comment]
Ok- added support for .gyp & CMake. FWIW it looks like the "HAVE_SYS_AHAFS_EVPRODS_H" check will also be broken in CMake.

It's a fairly unusual choice for a project to support 3 build systems. I certainly hope you don't plan to move away from autoconf. This would make things tricky for us because CMake depends on libuv to build itself so it's not clear how we would build libuv in the first place without autoconf.

**Comment by matthewbauer** [comment]
Here is the output for hrtime:

```
not ok 98 - hrtime
# exit code 6
# Output from process `hrtime`:
# Assertion failed in test/test-hrtime.c on line 50: diff < (uint64_t) 80 * NANOSEC / MILLISEC
```

**Comment by dtzWill** [comment]
FWIW this no longer applies cleanly (as github informs us), which I mention because I ran into this updating our (=Nixpkgs) version to the new 1.24.0 release where the patch no longer applies.  Not a blocker for me but pinging this PR so at least you know :).

**Comment by matthewbauer** [comment]
Ok. It is probably best to drop this patch and go back to the old way of providing the frameworks. I was hoping this would get merged so we wouldn’t have to deal with these conflicts. Now that it looks like this won’t be merged it is probably for the best to drop the patch.

**Comment by cjihrig** [comment]
I'll go ahead and close this, as it seems to have stalled out. Thanks for the PR anyway!

**Pull request closed without merging**